x-pack/plugins/security_solution/public/attack_discovery/pages/helpers.ts

```diff
@@ -73,6 +73,22 @@ export const getErrorToastText = (
   }
 };
 
+export const showNoAlertsPrompt = ({
+  alertsContextCount,
+  isLoading,
+}: {
+  alertsContextCount: number | null;
+  isLoading: boolean;
+}): boolean => !isLoading && alertsContextCount != null && alertsContextCount === 0;
```

Comment:
nit: Slight flicker of the underlying loading state when there are no alerts. Would need to have a slight delay in displaying the loader to avoid, which would be noticeable in all other generations, so probably fine to leave as-is unless we can check for alert counts client-side/separately.

<p align="center">
  <img width="500" src="https://github.com/elastic/kibana/assets/2946766/32e60cea-80cb-496b-836a-256e817cd2ae" />
</p>

Reply:
> nit: Slight flicker of the underlying loading state when there are no alerts. Would need to have a slight delay in displaying the loader to avoid, which would be noticeable in all other generations, so probably fine to leave as-is unless we can check for alert counts client-side/separately.

The alerts count is not (currently) available on the client side. The breif flash happens because there's a round trip to the server, which in-turn runs the query that retrieves alerts. A polling / caching implementation that makes the alerts count available client side would improve the current UX, but that feels a bit out of scope in this PR.